datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -170,4 +203,79 @@ int findTaskWithRoomForAPartition(List<String> tasks, Map<String, Set<String>> p
     }
     throw new DatastreamRuntimeException("No tasks found that can host an additional partition");
   }
+
+  void updateMetricsForDatastream(String datastream, PartitionAssignmentStats stats) {
+    if (!_partitionAssignmentStatsMap.containsKey(datastream)) {
+      registerLoadBasedPartitionAssignmentMetrics(datastream);
+    }
+    _partitionAssignmentStatsMap.put(datastream, stats);
+  }
+
+  private void registerLoadBasedPartitionAssignmentMetrics(String datastream) {
+    Supplier<Integer> minPartitionsAcrossTasksSupplier = () -> _partitionAssignmentStatsMap
+        .getOrDefault(datastream, PartitionAssignmentStats.DEFAULT).getMinPartitionsAcrossTasks();
+    DYNAMIC_METRICS_MANAGER.registerGauge(CLASS_NAME, datastream, MIN_PARTITIONS_ACROSS_TASKS,
+        minPartitionsAcrossTasksSupplier);
+
+    Supplier<Integer> maxPartitionsAcrossTasksSupplier = () -> _partitionAssignmentStatsMap
+        .getOrDefault(datastream, PartitionAssignmentStats.DEFAULT).getMaxPartitionsAcrossTasks();
+    DYNAMIC_METRICS_MANAGER.registerGauge(CLASS_NAME, datastream, MAX_PARTITIONS_ACROSS_TASKS,
+        maxPartitionsAcrossTasksSupplier);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<BrooklinMetricInfo> getMetricInfos() {
+    String prefix = CLASS_NAME + MetricsAware.KEY_REGEX;
+    return Arrays.asList(new BrooklinGaugeInfo(prefix + MIN_PARTITIONS_ACROSS_TASKS),
```

Comment:
Why is it better than what I'm doing? The strategy class exposes it as an unmodifiable list anyway (if that's the concern).

Reply:
Yes, Collections.unmodifiableList was the concern. I agree that the strategy class can take care of this. But, since this class is public and can be reused in the future, it is a better practice to ensure that we do most of the damage control in the class Library itself rather than relying on the callers to take care of it. We have seen in the past where the list was not suppose to be modified and the caller modifies it and the issue was super difficult to catch.